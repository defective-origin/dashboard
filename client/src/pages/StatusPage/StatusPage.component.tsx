import React from 'react'

// ---| core |---
import { useLocaleProvider, RouteProps } from 'Launcher'

// ---| pages |---
// ---| screens |---
// ---| components |---
import Banner from 'components/Banner'
import { Page } from 'components/Page'

// ---| common |---
import { cn } from 'common/tools'

// ---| self |---
import css from './StatusPage.module.scss'
import { STATUS_MAP, StatusType } from './StatusPage.conf'

export type StatusPageProps = RouteProps & {
  className?: string
  children?: React.ReactNode
  type?: StatusType
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <StatusPage />
 */
export function StatusPage(props: StatusPageProps): JSX.Element {
  const { type = 'default', children, className, ...otherProps } = props
  const locale = useLocaleProvider()
  const _className = cn(css.StatusPage, className)
  const status = STATUS_MAP[type]

  return (
    <Page name={type.toString()} className={_className} {...otherProps}>
      <Banner
        className={css.Banner}
        imageType={status.image}
        title={locale.t(status.title)}
        subtitle={locale.t(status.subtitle)}
        text={locale.t(status.text)}
      >
        {children}
      </Banner>
    </Page>
  )
}

StatusPage.displayName = 'StatusPage'

export default StatusPage
