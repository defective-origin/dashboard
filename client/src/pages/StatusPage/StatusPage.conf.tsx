// ---| core |---
import { TranslateKeys } from 'locale'

// ---| components |---
import { ImageVariant } from 'components/Image'

export type StatusVariant = 'welcome' | 401 | 403 | 404 | 500 | 502 | 503 | 'default'

export const STATUS_MAP: Record<StatusVariant, {
  image: ImageVariant,
  title: TranslateKeys,
  subtitle: TranslateKeys,
  text?: TranslateKeys,
}> = {
  welcome: {
    image: 'welcome',
    title: 'STATUS.WELCOME.TITLE',
    subtitle: 'STATUS.WELCOME.SUBTITLE',
    text: 'STATUS.WELCOME.TEXT',
  },
  401: {
    image: 400,
    title: 'STATUS.401.TITLE',
    subtitle: 'STATUS.401.SUBTITLE',
  },
  403: {
    image: 400,
    title: 'STATUS.403.TITLE',
    subtitle: 'STATUS.403.SUBTITLE',
  },
  404: {
    image: 400,
    title: 'STATUS.404.TITLE',
    subtitle: 'STATUS.404.SUBTITLE',
  },
  500: {
    image: 500,
    title: 'STATUS.500.TITLE',
    subtitle: 'STATUS.500.SUBTITLE',
  },
  502: {
    image: 500,
    title: 'STATUS.502.TITLE',
    subtitle: 'STATUS.502.SUBTITLE',
  },
  503: {
    image: 500,
    title: 'STATUS.503.TITLE',
    subtitle: 'STATUS.503.SUBTITLE',
  },
  default: {
    image: 500,
    title: 'STATUS.DEFAULT.TITLE',
    subtitle: 'STATUS.DEFAULT.SUBTITLE',
  },
}
