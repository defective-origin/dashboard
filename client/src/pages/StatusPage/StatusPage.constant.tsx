import { BannerProps } from 'components/Banner'

export type StatusType = 0 | 401 | 403 | 404 | 500 | 502 | 503 | 'default'

const image500 = 'https://d1k5j68ob7clqb.cloudfront.net/processed/with_watermark/6d3H7wV3NUrpA7.png'
const image400 = 'https://png.pngtree.com/png-vector/20230416/ourmid/pngtree-unicorn-full-body-beautiful-pattern-png-image_6704420.png'
export const STATUS_MAP: Record<StatusType, BannerProps> = {
  0: {
    src: 'https://www.dndbeyond.com/avatars/thumbnails/30836/227/1000/1000/638063931763028274.png',
    title: `WELCOME TO OUR APP!`,
    subtitle: 'Please read the guide before start work.',
    text: 'insert button links to guide and dashboard page',
  },
  401: {
    src: image400,
    title: 'NOT AUTHORIZED!',
    subtitle: `Sorry but it seems like we can't let you in.`,
  },
  403: {
    src: image400,
    title: 'YOU HAVE NO PERMISSION!',
    subtitle: `Sorry but it seems like we can't let you in.`,
  },
  404: {
    src: image400,
    title: `OOOOPS! WE COULDN'T FIND THIS PAGE!`,
    subtitle: 'It could be deleted or never existed.',
  },
  500: {
    src: image500,
    title: '500 ERROR! SOMETHING WENT WRONG!',
    subtitle: 'TEXT_500',
  },
  502: {
    src: image500,
    title: 'BAD GATEWAY!',
    subtitle: 'TEXT_500',
  },
  503: {
    src: image500,
    title: 'SERVICE UNAVAILABLE!',
    subtitle: 'TEXT_500',
  },
  default: {
    src: image500,
    title: 'SOMETHING WENT WRONG!',
    subtitle: `Looks like we are having some issues. We'll get it fixed as soon as possible. Please try again later.`,
  },
}
