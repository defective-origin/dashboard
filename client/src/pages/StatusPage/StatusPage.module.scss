@import 'src/App/App.module';

:global {
  .dark {
    // --text-color: white;
  }

  .light {
    // --text-color: black;
  }
}

.StatusPage {
  position: relative;
}

.Banner {
  width: 450px;
  position: absolute;

  // todo: fix via @include absolute-position(center, center);

  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
}
