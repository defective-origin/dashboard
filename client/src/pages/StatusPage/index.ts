/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './StatusPage.component'

/* INJECT_EXPORT_PLACE */
export * from './StatusPage.component'
