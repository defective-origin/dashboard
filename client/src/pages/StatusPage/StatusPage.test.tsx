// ---| tests |---
import { render } from 'tests'

// ---| self |---
import StatusPage from './StatusPage.component'

describe('[StatusPage] component', () => {
  it('should render component', () => {
    const container = render(<StatusPage />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})
