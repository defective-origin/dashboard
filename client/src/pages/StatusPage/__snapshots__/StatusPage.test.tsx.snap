// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[StatusPage] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="layout layout--row page _StatusPage_eba8ea"
  >
    <div
      class="layout-item layout-item--content"
    >
      <div
        class="_Banner_fd7a36 _Banner_eba8ea"
      >
        <img
          class="image _Image_fd7a36"
          src="https://d1k5j68ob7clqb.cloudfront.net/processed/with_watermark/6d3H7wV3NUrpA7.png"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
